addons/website_forum/static/src/components/flag_mark_as_offensive/flag_mark_as_offensive.js

```diff
@@ -0,0 +1,32 @@
+/** @odoo-module **/
+
+import { Component, useEffect } from "@odoo/owl";
+import { useChildRef } from "@web/core/utils/hooks";
+import { Dialog } from "@web/core/dialog/dialog";
+
+export class FlagMarkAsOffensiveDialog extends Component {
```

Comment:
FlagAsOffensive

Reply:
No, the post is here already flagged and this is used to convert it to offensive.
Maybe `convertFlaggedToOffensive` if you do want to rename it.

Reply:
My mistake, seemed to be redundant at first glance.